src/handlers.ts

```diff
@@ -375,71 +377,102 @@ export async function siteChangedWebhook(
     });
 }
 
+async function buildInboxItem(
+    uri: string,
+    db: KvStore,
+    apCtx: APContext<ContextData>,
+    liked: string[] = [],
+): Promise<InboxItem | null> {
+    const item = await db.get<InboxItem>([uri]);
+
+    // If the item is not in the db, return null as we can't build it
+    if (!item) {
+        return null;
+    }
+
+    // If the object associated with the item is a string, it's probably a URI,
+    // so we should look it up in the db. If it's not in the db, we should just
+    // leave it as is
+    if (typeof item.object === 'string') {
+        item.object = await db.get([item.object]) ?? item.object;
+    }
+
+    // If the object associated with the item is an object with a content property,
+    // we should sanitize the content to prevent XSS (in case it contains HTML)
+    if (item.object && typeof item.object !== 'string' && item.object.content) {
+        item.object.content = sanitizeHtml(item.object.content, {
+            allowedTags: ['a', 'p', 'img', 'br', 'strong', 'em', 'span'],
+            allowedAttributes: {
+                a: ['href'],
+                img: ['src'],
+            }
+        });
+    }
+
+    // If the associated object is a Like, we should check if it's in the provided
+    // liked list and add a liked property to the item if it is
+    let objectId: string = '';
+
+    if (typeof item.object === 'string') {
+        objectId = item.object;
+    } else if (typeof item.object.id === 'string') {
+        objectId = item.object.id;
+    }
+
+    if (objectId) {
+        const likeId = apCtx.getObjectUri(Like, {
+            id: createHash('sha256').update(objectId).digest('hex'),
+        });
+        if (liked.includes(likeId.href)) {
+            if (typeof item.object !== 'string') {
+                item.object.liked = true;
+            }
+        }
+    }
+
+    // Return the built item
+    return item;
+}
+
 export async function inboxHandler(
```

Comment:
The diff for this is a little hard to read - The changes make this essentially the same as the implementation for `getActivities` just without any pagination